### PR TITLE
Update featured-resource component to display both internal and external links

### PR DESCRIPTION
**Pull request by nick-mon1**
## Summary
Enables displaying external links for the `core/featured-resources.html` component.
Component currently only supports hugo internal pages.

### Preview

- [HCD Guide](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-enhance-featured-resource/guides/hcd/design-concepts/)
- [Analytics Topic](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-enhance-featured-resource/topics/analytics)

Screenshot 2024-08-20 at 1 49 03 PM

### Solution

- Simplified `core/featured-resource.html` API to handle both internal and external links (801c293b61f4a1a17619bd9387f0111948b068a2)
- Updated short code to have same component API (350b8d2618f2b950da5d41918edf68222feb3b63)

#### What to test

| test case             | passing |
| --------------------- | ------- |
| internal link         | yes     |
| external link (https) | yes     |
| external link (http)  | yes     |
| multiple resources    | yes     |
| shortcode | yes |

1. **Topics Frontmatter** update `yaml` in topics page front matter, use `content/topics/analytics/_index.md` to test

**internal link**

```yaml
featured_resources:
  resources:
  - link: "/an-introduction-to-accessibility"
```

**external link**

```yaml
featured_resources:
  resources:
  - link: "https://www.whitehouse.gov/wp-content/uploads/legacy_drupal_files/omb/memoranda/2010/m10-23.pdf"
    title: "External website title goes here"
    summary: "Short summary goes here for the reader to understand the nature of the external content."
    kicker: "Custom kicker"
```

**multiple resources** should display in order

```yaml
featured_resources:
  resources:
  - link: "/guides/dap"
  - link: "http://www.whitehouse.gov/wp-content/uploads/legacy_drupal_files/omb/memoranda/2010/m10-23.pdf"
    title: "External website title goes here"
    summary: "Short summary goes here for the reader to understand the nature of the external content."
    kicker: "Custom kicker"
  - link: "http://www.whitehouse.gov/wp-content/uploads/legacy_drupal_files/omb/memoranda/2010/m10-23.pdf"
    title: "External website title goes here"
    summary: "Short summary goes here for the reader to understand the nature of the external content."
    kicker: "Custom kicker"
```

2. **Shortcode** use `content/guides/hcd/design-operations/_index.md` to test

**internal link**

```html
{{< featured-resource link = "/guides/hcd/design-concepts">}}
```

**external link**

```html
{{< featured-resource link = "https://www.ai.gov" kicker = "Visit ai.gov" title = "Check out ai.gov" summary = "ai.gov is the go to reference for all artifical intelligence in the federal government" >}}
```

---

### Dev Checklist

- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:13]: Are these missing params missing their respective comments i.e. `@params {TYPE}`?

Or are they coming from the `resource_data` object?

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:32]: Indents look off on the rest of these examples.

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:23]: Can we remove these unnecessary spaces from examples?

```suggestion
  {{< featured-resource link="/guides/hcd/discovery-concepts" >}}
```

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:32]: This uses the `-` to note its a list, but it's all inline. Can we wither use commas for an inline list or break these out into their own line?

Ex: `allowed paths: communicators, multilingual, plain-language, etc etc`.

Alternatively:

```md
allowed paths: 
  - communicators
  - multilingual
  - plain-language
```

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:44]: Is there an example of a featured resource that's an external link?

Wasn't the original intent of this component to feature our internal content?

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:75]: Can we set this `href` in a variable instead? The logic inside makes it hard to read.

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:79]: While this does make it more readable; it also ties it too much to the text class.

Meaning, if we wanted to create a global kicker style, we couldn't do that with this component.

I'd recommend something like example below. That way it's a little more flexible in its placement.

```suggestion
        <p class="dg-featured-resource__kicker">
```

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:90]: If `$title` is missing it'll still render the `<h2>`.

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:95]: Same with summary.

```suggestion
        {{ if $summary }}
          <p class="dg-featured-resource__text-description">
              {{- $summary -}}
          </p>
        {{ end }}
```

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:99]: We're always passing `.Site` to this component. 

Anyway to make that the default instead of passing the same param every time?

File: themes/digital.gov/layouts/shortcodes/featured-resource.html
Comment (mejiaj) [inline themes/digital.gov/layouts/shortcodes/featured-resource.html:1]: You don't need to duplicate comments if it relies on the core partial. You can add a note to view `partials/core/featured-resource.html`.

More useful to add examples of how to use or anything specific to shortcode format.

**Review by mejiaj — CHANGES_REQUESTED** [review]
No major issues found. Small suggestions added.

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (RileySeaburg) [reply themes/digital.gov/layouts/partials/core/featured-resource.html:44]: It would be nice to have both and declaratively switch from one to the other.

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (RileySeaburg) [reply themes/digital.gov/layouts/partials/core/featured-resource.html:79]: Are Kicker styles really necessary in this project?

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (RileySeaburg) [inline themes/digital.gov/layouts/partials/core/featured-resource.html:86]: Please move this if above the h2

**Review by RileySeaburg — CHANGES_REQUESTED** [review]
Just fix the logic that is rendering an h2 if nothing appears.

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/featured-resource.html:13]: I've updated the comments to explain they are all part of the same `resource_data` object.

File: content/topics/analytics/_index.md
Comment (nick-mon1) [inline content/topics/analytics/_index.md:32]: ### Fix

Update wording to be more clear

`Featured resource at the top of the page`

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/featured-resource.html:44]: The requirements for this component has shifted, the team is okay with using it for both internal and external links.

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/featured-resource.html:99]: Good call out, but we are stuck with having to pass in `.Site` to partials/components with the current code.

We could refactor this partial to re-reference the global scope:

```go

{{ $site := . }}

```

This would require additional refactoring if we should pursue this.

**Comment by RileySeaburg** [comment]
Is this still Hold?

File: themes/digital.gov/layouts/partials/core/featured-resource.html
Comment (mejiaj) [reply themes/digital.gov/layouts/partials/core/featured-resource.html:86]: @nick-mon1 leaving as-is could result in an empty title. Even it _always_ showed, we should be clear in the intent.

**Review by mejiaj — COMMENTED** [review]
@nick-mon1 two minor comments weren't resolved. Can you address those and we can get this merged?

**Review by RileySeaburg — APPROVED** [review]
This looks great, do we want to wait for @bonnieAcameron to make those updates before we merge? @nick-mon1

**Comment by RileySeaburg** [comment]
@mejiaj please confirm you still need these changes addresses by doing another review, it's difficult to determine if @nick-mon1 has addressed them.

**Comment by bonnieAcameron** [comment]
I have updated the metadata for those 3 topic pages, @nick-mon1 @RileySeaburg

**Comment by nick-mon1** [comment]
@bonnieAcameron Thanks for the updates.
@RileySeaburg and @mejiaj I've resolved the outstanding comments.

**Review by mejiaj — APPROVED** [review]
Comments have been addressed. Thanks @nick-mon1.

**Review by RileySeaburg — APPROVED** [review]
Great job on this

**Pull request merged**
